Added coverage for 'Anonymous' feature

alliefeld opened the pull request:
Resolves: #24 
PR overview:
Wrote a built-in test to prove that the is_anonymous tag is a valid property of posts that can be saved and retrieved.

Changes made:
'test/src/posts.js'

Testing:
Ensuring coverage % increased and all tests written run without crashing.

alliefeld committed:
something

alliefeld committed:
Removed commented code

alliefeld committed:
Linting error

mmwathe reviewed (changes requested):
@rmdluo  When we create the post we should expect the anonymous status to default to false, so we should either set the is_anonymous to false or not set the field in the before block

rmdluo committed:
added test to check if newly created posts default is_anon to false

removed is_anonymous from before() post creation, added it() to check
that it defaulted to false

mmwathe commented:
Looks good! Adds testing for default status of post and sending request to database updates the anonymous field accordingly.
